File: fixtures/java/type_inference/overloads/Overloads.java
package app;

public class Overloads {
    public int combine(int a, int b) {
        return a + b;
    }

    public String combine(String a, String b) {
        return a + b;
    }

    public String combine(int a, String b) {
        return a + b;
    }

    public String run() {
        int ints = combine(1, 2);
        String strs = combine("a", "b");
        String mixed = combine(1, "b");
        return ints + strs + mixed;
    }
}

File: fixtures/scala/type_inference/overloads/Overloads.scala
package app

object Overloads {
  def combine(a: Int, b: Int): Int = a + b

  def combine(a: String, b: String): String = a + b

  def combine(a: Int, b: String): String = s"$a$b"

  def run(): String = {
    val ints = combine(1, 2)
    val strs = combine("a", "b")
    val mixed = combine(1, "b")
    s"$ints$strs$mixed"
  }
}
